docs/docs/api-ref.md

```diff
@@ -5,257 +5,3010 @@ layout: docs
 
-Populates a preview image for a given workbook. You must populate the image before you can iterate through the
-connections.
+
+
+**Attributes**
+
+Name | Description  
+:--- | :--- 
+`Field` | Defined in the `RequestOptions.Field` class.
+`Operator` | Defined in the `RequestOptions.Operator` class
+`Value` | The value to compare with the specified field and operator.
+
+
+
+
+
+<br>  
+<br>   
+
+
+## Groups
+
+Using the TSC library, you can get information about all the groups on a site, you can add or remove groups, or add or remove users in a group.
+
+The group resources for Tableau Server are defined in the `GroupItem` class. The class corresponds to the group resources you can access using the Tableau Server REST API. The group methods are based upon the endpoints for groups in the REST API and operate on the `GroupItem` class.
+
+<br>   
+<br> 
+
+### GroupItem class
 
 ```py
-Workbooks.populate_connections(workbook_obj)
+GroupItem(name)
 ```
 
-### Get Views for a Workbook
+The `GroupItem` class contains the members or attributes for the view resources on Tableau Server. The `GroupItem` class defines the information you can request or query from Tableau Server. The class members correspond to the attributes of a server request or response payload.
 
-Returns a list of views for a workbook. Before you get views, you must call populate_views.
+Source file: models/group_item.py
 
+**Attributes**
+
+Name | Description  
+:--- | :--- 
+`domain_name` |  The name of the Active Directory domain (`local` if local authentication is used).  
+`id` |  The id of the group.  
+`users`  |   The list of users (`UserItem`).  
+`name` |  The name of the group.  The `name` is required when you create an instance of a group.
+
+
+
+**Example**
+
+```py
+ newgroup = TSC.GroupItem('My Group')
 ```
-workbook_obj.views
+
+
+
+
+<br>   
+<br> 
+
+### Groups methods
+
+The Tableau Server Client provides several methods for interacting with group resources, or endpoints. These methods correspond to endpoints in the Tableau Server REST API.
+
+
+
+Source file: server/endpoint/groups_endpoint.py
+
+<br>   
+<br> 
+
+#### groups.add_user
+
+```py
+groups.add_user(group_item, user_id):
 ```
 
-### Get Connections for a Workbook
+Adds a user to the specified group. 
+
+
+REST API [Add User to Group](http://onlinehelp.tableau.com/current/api/rest_api/en-us/help.htm#REST/rest_api_ref.htm#Add_User_to_Group%3FTocPath%3DAPI%2520Reference%7C_____8){:target="_blank"}
+
+**Parameters**
+
+Name | Description  
+:--- | :--- 
+`group_item`  | The `group_item` specifies the group to update.
+`user_id` | The id of the user. 
+
+
+
+
+**Returns**
+
+None.    
+
+
+**Example**
+
+```py
+# Adding a user to a group
+# Using the second group on the site, aleady have all_groups
+# The id for Ian is '59a8a7b6-be3a-4d2d-1e9e-08a7b6b5b4ba'
+
+# add Ian to the second group
+  server.groups.add_user(all_groups[1], '59a8a7b6-be3a-4d2d-1e9e-08a7b6b5b4ba')
```

Comment:
Whenever possible we should avoid the [subscript] notation in example because it's not obvious what it means. We can always pop it from the list and give it a name